xds/src/main/java/io/grpc/xds/ClusterResolverLoadBalancer.java

```diff
@@ -0,0 +1,684 @@
+        logger.log(XdsLogLevel.INFO, "Start watching EDS resource {0}", resourceName);
+        xdsClient.watchEdsResource(resourceName, this);
+      }
+
+      @Override
+      protected void shutdown() {
+        super.shutdown();
+        String resourceName = edsServiceName != null ? edsServiceName : name;
+        logger.log(XdsLogLevel.INFO, "Stop watching EDS resource {0}", resourceName);
+        xdsClient.cancelEdsResourceWatch(resourceName, this);
+      }
+
+      @Override
+      public void onChanged(final EdsUpdate update) {
+        class EndpointsUpdated implements Runnable {
+          @Override
+          public void run() {
+            if (shutdown) {
+              return;
+            }
+            logger.log(XdsLogLevel.DEBUG, "Received endpoint update {0}", update);
+            if (logger.isLoggable(XdsLogLevel.INFO)) {
+              logger.log(XdsLogLevel.INFO, "Cluster {0}: {1} localities, {2} drop categories",
+                  update.getClusterName(), update.getLocalityLbEndpointsMap().size(),
+                  update.getDropPolicies().size());
+            }
+            Map<Locality, LocalityLbEndpoints> localityLbEndpoints =
+                update.getLocalityLbEndpointsMap();
+            List<DropOverload> dropOverloads = update.getDropPolicies();
+            List<EquivalentAddressGroup> addresses = new ArrayList<>();
+            Map<String, Map<Locality, Integer>> prioritizedLocalityWeights = new HashMap<>();
+            for (Locality locality : localityLbEndpoints.keySet()) {
+              LocalityLbEndpoints localityLbInfo = localityLbEndpoints.get(locality);
+              int priority = localityLbInfo.getPriority();
+              String priorityName = priorityName(name, priority);
+              boolean discard = true;
+              for (LbEndpoint endpoint : localityLbInfo.getEndpoints()) {
+                if (endpoint.isHealthy()) {
+                  discard = false;
+                  EquivalentAddressGroup eag =
+                      AddressFilter.setPathFilter(
+                          endpoint.getAddress(),
+                          Arrays.asList(priorityName, localityName(locality)));
+                  addresses.add(eag);
+                }
+              }
+              if (discard) {
+                logger.log(XdsLogLevel.INFO,
+                    "Discard locality {0} with 0 healthy endpoints", locality);
+                continue;
+              }
+              if (!prioritizedLocalityWeights.containsKey(priorityName)) {
+                prioritizedLocalityWeights.put(priorityName, new HashMap<Locality, Integer>());
+              }
+              prioritizedLocalityWeights.get(priorityName).put(
+                  locality, localityLbInfo.getLocalityWeight());
+            }
+            if (prioritizedLocalityWeights.isEmpty()) {
+              // Transient state, neither as a result or an error.
```

Comment:
Why this change? I might have missed a discussion about it.

Reply:
Changed to still propagate the update. It will behave the same as the EDS resource does not exist.

The desired behavior is still questionable:

When receiving an EDS update that contains priorities/localities but no usable endpoints (in reality we've seen this happen transiently during TD resource migration):

| previously has usable endpoints | other clusters have usable endpoints | what to propagate to child LB policy |
| -------------- | ----------- | ----------- |
| True      | True         | Include endpoint from previous update? Or __just endpoints from other discovered clusters__? |
| True      | False        | Keep buffering RPCs if some cluster has not been discovered. Otherwise, TF and fail RPCs |
| False     | True         | Just endpoints from other discovered clusters |
| False     | False        | (If all clusters have been discovered) Keep buffering RPCs? Or __TF and fail RPCs__? |

Highlighted behavior is the implementation in this PR.